src/titiler/extensions/tests/test_wms.py

```diff
@@ -0,0 +1,373 @@
+                "LAYERS": cog,
+                "rescale": "0,1000",
+            },
+        )
+        assert response.status_code == 200
+        assert "rescale=0%2C1000" in response.text
+
+        response = client.get(
+            "/wms",
+            params={
+                "REQUEST": "GetCapabilities",
+                "SERVICE": "wms",
+                "LAYERS": f"{cog1},{cog2}",
+            },
+        )
+        assert response.status_code == 200
+        root = ET.fromstring(response.text)
+        assert len(root.find("Capability").find("Layer").findall("Layer")) == 2
+        assert cog1 in response.text
+        assert cog2 in response.text
+
+
+def test_wmsExtension_GetMap():
+    """Test wmsValidateExtension class."""
+    tiler_plus_wms = TilerFactory(extensions=[wmsExtension()])
+
+    app = FastAPI()
+    app.include_router(tiler_plus_wms.router)
+    with TestClient(app) as client:
+        response = client.get(
+            "/wms",
+            params={
+                "VERSION": "1.3.0",
+                "REQUEST": "GetMap",
+                "LAYERS": cog,
+            },
+        )
+        assert response.status_code == 400
+        assert "Missing 'GetMap' parameters: " in response.text
+
+        response = client.get(
+            "/wms",
+            params={
+                "VERSION": "1.3.0",
+                "REQUEST": "GetMap",
+                "LAYERS": cog,
+                "BBOX": "373185.0,8019284.949381611,639014.9492102272,8286015.0",
+                "CRS": "EPSG:32621",
+                "WIDTH": 334,
+                "HEIGHT": 333,
+                "FORMAT": "image/png",
+                "TRANSPARENT": False,
+            },
+        )
+        assert response.status_code == 200
+        meta = parse_img(response.content)
+        assert meta["driver"] == "PNG"
+        assert meta["width"] == 334
+        assert meta["height"] == 333
+        assert meta["count"] == 1
+
+        response = client.get(
+            "/wms",
+            params={
+                "VERSION": "1.3.0",
+                "REQUEST": "GetMap",
+                "LAYERS": cog,
+                "BBOX": "373185.0,8019284.949381611,639014.9492102272,8286015.0",
+                "CRS": "EPSG:32621",
+                "WIDTH": 334,
+                "HEIGHT": 333,
+                "FORMAT": "image/png",
+                "TRANSPARENT": True,
+            },
+        )
+        assert response.status_code == 200
+        meta = parse_img(response.content)
+        assert meta["driver"] == "PNG"
+        assert meta["width"] == 334
+        assert meta["height"] == 333
+        assert meta["count"] == 2
+
+        response = client.get(
+            "/wms",
+            params={
+                "VERSION": "1.3.0",
+                "REQUEST": "GetMap",
+                "LAYERS": cog,
+                "BBOX": "373185.0,8019284.949381611,639014.9492102272,8286015.0",
+                "SRS": "EPSG:32621",
```

Comment:
WMS 1.3.0 uses `crs` instead of `srs`. Annoying but small change. Geoserver will accept either just for the sake of flexibility. I suggest we stick with interchangeable SRS/CRS, unless you want to go by the OGC spec. For reference: https://docs.geoserver.org/stable/en/user/services/wms/basics.html

Reply:
I didn't know if this was version related. 👍 for supporting both whatever the version is